syntax = "proto2";

package encryptedIM;

message StoredPackage{
        required string nickname = 20;
        required string content = 30;
}

message EncryptedPackage{
    required bytes iv = 20;
	required bytes encryptedMessage = 30;
}

message PlaintextAndMAC {
	required bytes paddedPlaintext = 10;
	required bytes mac = 20;
}

message IM {
	required string nickname = 5;
	required string message = 10;
}
